### PR TITLE
Added input field to mutations

### LINKED ISSUE #93: Investigate advantages of one input field in the mutation
Related articles - https://atheros.ai/blog/graphql-best-practices-for-graphql-schema-design

### PR DESCRIPTION
**Pull request by mikhail-kilin**
### Summary

For some mutations added input objects

### How it works

The single advantage I found is more simple request from the client side. Now we need to form something like this:
```graphql
mutation SignUp($email: String!, $password: String!, $firstName: String, $lastName: String) {
  signup(email: $email, password: $password, firstName: $firstName, lastName: $lastName) {
    me {
      id
      email
    }
    accessToken
    refreshToken
  }
}
```

With `InputObject` the same request will look like this:
```graphql
mutation SignUp($input: SignupInput!) {
  signup(input: $input) {
    me {
      id
      email
    }
    accessToken
    refreshToken
  }
}
``` 
with query variables:
```graphql
{
  "input": {
	"email": "john.tolkien@example.com",
  	"password": "654321",
  	"firstName": "John",
  	"lastName": "Tolkien"
  }
}
```

### Test plan

None

### Review notes

While reviewing pull-request (especially when it's your pull-request),
please make sure that:

- you understand what problem is solved by PR and how is it solved
- new tests are in place, no redundant tests
- DB schema changes reflect new migrations
- newly introduced DB fields have indexes and constraints
- there are no missed files (migrations, view templates)
- required ENV variables added and described in `.env.example` and added to Heroku
- associated Heroku review app works correctly with introduced changes

### Deploy notes

As I think we need to notify frontend command about changes

### References
* Resolves #93 Issue

### CODE REVIEW

File: app/graphql/mutations/presign_data.rb [hunk 1]
```diff
@@ -1,12 +1,11 @@
...
-    argument :filename, String, required: true
-    argument :type, String, required: true
+    argument :input, Types::PresignDataInput, required: true
 
     type Types::PresignType
 
-    def resolve(filename:, type:)
-      presign_image = PreparePresignImage.call(filename: filename, type: type)
+    def resolve(input:)
+      presign_image = PreparePresignImage.call(input.to_hash)
```
Comment (EvgeniyEsaulkov) [inline app/graphql/mutations/presign_data.rb:8]: Can you explain please why did you use here implicit type conversion instead `.to_h`?

**Review by EvgeniyEsaulkov — CHANGES_REQUESTED** [review]
Would you like to add input object for `SignOut` mutation too?

File: app/graphql/mutations/presign_data.rb (hunk 1 above)
Comment (mikhail-kilin) [reply app/graphql/mutations/presign_data.rb:8]: It was my mistake)

**Comment by ArthurZaharov** [comment]
@mikhail-kilin this PR will break FE application. Could you coordinate it with them?

File: spec/graphql/mutations/update_password_spec.rb [hunk 2]
```diff
@@ -3,13 +3,18 @@
 describe Mutations::UpdatePassword do
   include_context "when time is frozen"
 
-  let(:user) { create(:user, :with_reset_token, email: "john.doe@example.com", first_name: "John", last_name: "Doe") }
+  let(:user) do
+    create(:user, :with_reset_token, id: 111_111, email: "john.doe@example.com", first_name: "John", last_name: "Doe")
```
Comment (EvgeniyEsaulkov) [inline spec/graphql/mutations/update_password_spec.rb:7]: As we specified user id here I guess we don't need to call `gsub` method for fixture file:
https://github.com/fs/rails-base-graphql-api/blob/7c25ca8e4f945ffe0dbe70813a3c10ef94ab080e/spec/graphql/mutations/update_password_spec.rb#L37-L39
Instead this we can just hardcode user id within the file

**Review by EvgeniyEsaulkov — CHANGES_REQUESTED** [review]

File: spec/graphql/mutations/update_password_spec.rb (hunk 2 above)
Comment (KilinMikhail) [reply spec/graphql/mutations/update_password_spec.rb:7]: fixed

**Review by EvgeniyEsaulkov — APPROVED** [review]

**Pull request merged**